src/main/java/co/rsk/bitcoinj/script/P2shErpFederationRedeemScriptParser.java

```diff
@@ -1,29 +1,75 @@
 package co.rsk.bitcoinj.script;
 
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.Sha256Hash;
 import co.rsk.bitcoinj.core.Utils;
 import co.rsk.bitcoinj.core.VerificationException;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class P2shErpFederationRedeemScriptParser extends StandardRedeemScriptParser {
+public class P2shErpFederationRedeemScriptParser implements RedeemScriptParser {
     private static final Logger logger = LoggerFactory.getLogger(P2shErpFederationRedeemScriptParser.class);
 
     public static long MAX_CSV_VALUE = 65_535L; // 2^16 - 1, since bitcoin will interpret up to 16 bits as the CSV value
 
+    private final MultiSigType multiSigType;
+    private final RedeemScriptParser redeemScriptParser;
```

Comment:
Should we rename this to standardRedeemScriptParser maybe?

Reply:
Solved